lib/gcloud/logging/project.rb

```diff
@@ -0,0 +1,362 @@
+        else
+          fail ApiError.from_response(resp)
+        end
+      end
+      alias_method :find_resources, :resources
+
+      ##
+      # Retrieves the list of sinks belonging to the project.
+      #
+      # @param [String] token A previously-returned page token representing part
+      #   of the larger set of results to view.
+      # @param [Integer] max Maximum number of sinks to return.
+      #
+      # @return [Array<Gcloud::Logging::Sink>] (See
+      #   {Gcloud::Logging::Sink::List})
+      #
+      # @example
+      #   require "gcloud"
+      #
+      #   gcloud = Gcloud.new
+      #   logging = gcloud.logging
+      #   sinks = logging.sinks
+      #   sinks.each do |sink|
+      #     puts sink.name
+      #   end
+      #
+      # @example With pagination: (See {Gcloud::Logging::Sink::List})
+      #   require "gcloud"
+      #
+      #   gcloud = Gcloud.new
+      #   logging = gcloud.logging
+      #   sinks = logging.sinks
+      #   loop do
+      #     sinks.each do |sink|
+      #       puts sink.name
+      #     end
+      #     break unless sinks.next?
+      #     sinks = sinks.next
+      #   end
+      #
+      def sinks token: nil, max: nil
+        ensure_connection!
+        resp = connection.list_sinks token: token, max: max
+        if resp.success?
+          Sink::List.from_response resp, connection
+        else
+          fail ApiError.from_response(resp)
+        end
+      end
+      alias_method :find_sinks, :sinks
+
+      ##
+      # Creates a new sink.
+      #
+      # @param [String] name The client-assigned sink identifier. Sink
+      #   identifiers are limited to 1000 characters and can include only the
+      #   following characters: `A-Z`, `a-z`, `0-9`, and the special characters
+      #   `_-.`.
+      # @param [String] destination The export destination. See [Exporting Logs
+      #   With
+      #   Sinks](https://cloud.google.com/logging/docs/api/tasks/exporting-logs).
+      # @param [String] filter An [advanced logs
+      #  filter](https://cloud.google.com/logging/docs/view/advanced_filters)
+      #  that defines the log entries to be exported. The filter must be
+      #  consistent with the log entry format designed by the `version`
+      #  parameter, regardless of the format of the log entry that was
+      #  originally written to Cloud Logging.
+      # @param [Symbol] version The log entry version used when exporting log
+      #   entries from this sink. This version does not have to correspond to
+      #   the version of the log entry when it was written to Cloud Logging.
+      #   Accepted values are `:unspecified`, `:v2`, and `:v1`.
+      #
+      # @return [Gcloud::Logging::Sink]
+      #
+      # @example
+      #   require "gcloud"
+      #
+      #   gcloud = Gcloud.new
+      #   logging = gcloud.logging
+      #   sink = logging.create_sink "my-sink"
+      #
+      def create_sink name, destination: nil, filter: nil, version: :unspecified
```

Comment:
It seems like the default behavior is when unspecified it defaults to V2. That could change over time, so the way I read it is that the default value is `unspecified` which will then become `V2`. So I thought it made sense to make the default `unspecified`. But right now I'm just trying to establish the service API defaults, we can always refine the ruby code defaults as we polish the ruby API. Getting some acceptance tests would help clarify this I bet.